### PR TITLE
Add missing changelog entries (since mid 2018)

### PR DESCRIPTION
**Pull request by ePaul**
We recently found that we didn't update the changelog for quite a while.
So I took some hours to go through the merged pull requests and add changelog entries for the ones which have guideline changes.

### CODE REVIEW

**Comment by tkrop** [comment]
@ePaul Many thanks. Looks like an excellent work!

**Comment by tkrop** [comment]
:+1:

File: chapters/changelog.adoc [hunk 1]
```diff
@@ -10,11 +10,47 @@ The latter changes are additionally labeled with "Rule Change" here.
...
 
+(Note that recent changes might be missing, as we update this list only occasionally,
+ not with each pull request, to avoid merge commits.)
+
 [[rule-changes]]
 == Rule Changes
 
-* `2019-01-24:` Improve guidance on caching (<<149>>, <<227>>). 
-* `2019-01-15:` Improve guidance on idempotency, introduce idempotency-key (<<229>>, <<231>>).
+* `2021-01-19`: more details for {GET-with-Body} and {DELETE-with-Body} (<<148>>).
```
Comment (tfrauenstein) [inline chapters/changelog.adoc:19]: --> standards for specifying {GET-with-Body} and {DELETE-with-Body} (<<148>>).

File: chapters/changelog.adoc [hunk 2]
```diff
@@ -10,11 +10,47 @@ The latter changes are additionally labeled with "Rule Change" here.
...
+* `2020-09-29`: include models for headers to be included by reference in API definitions (<<183>>)
```
Comment (tfrauenstein) [inline chapters/changelog.adoc:20]: --> specify headers of endpoints via reference to standard definitions in <<183>>.

File: chapters/changelog.adoc [hunk 3]
```diff
@@ -10,11 +10,47 @@ The latter changes are additionally labeled with "Rule Change" here.
...
+* `2020-09-08`: add exception for legacy host names to <<224>>
+* `2020-08-25`: change <<240>> from {MUST} to {SHOULD}, explain exceptions
+* `2020-08-25`: add exception for `self` to <<143>> and <<134>>.
```
Comment (tfrauenstein) [inline chapters/changelog.adoc:23]: --> add `self` pseudo-identifier as exception to <<143>> and <<134>>.

File: chapters/changelog.adoc [hunk 4]
```diff
@@ -10,11 +10,47 @@ The latter changes are additionally labeled with "Rule Change" here.
...
+* `2020-08-24`: change "{MUST} avoid trailing slashes" to <<136>>.
```
Comment (tfrauenstein) [inline chapters/changelog.adoc:24]: --> standardise usage of slashes in <<136>>.

File: chapters/changelog.adoc [hunk 5]
```diff
@@ -10,11 +10,47 @@ The latter changes are additionally labeled with "Rule Change" here.
...
+* `2020-02-05`: add Sunset header, clarify deprecation producedure (<<185>>, <<186>>, <<187>>, <<188>>, <<189>>, <<190>>, <<191>>)
+* `2020-01-21`: new rule <<240>> (as MUST, changed later to SHOULD)
+* `2020-01-15`: change "Warning" to "Deprecation" header in <<189>>, <<190>>.
+* `2019-10-10`: remove never-implemented rule "{MUST} Permissions on events must correspond to API permissions"
+* `2019-09-10`: remove duplicated rule "{MAY} Standards could be used for Language, Country and Currency", upgrade <<170>> from {MAY} to {SHOULD}.
+* `2019-08-29`: new rule <<239>>, extend <<167>> pointing to {RFC-7493}[RFC-7493]
+* `2019-08-29`: new rules <<236>>, <<237>>
+* `2019-07-30`: new rule <<238>>
+* `2019-07-30`: change <<519>> from {SHOULD} to {MUST}
+* `2019-07-30`: change "{SHOULD} Null values should have their fields removed to" <<123>>.
```
Comment (tfrauenstein) [inline chapters/changelog.adoc:38]: --> update guideance for how to specify optional and nullable properties in  <<123>>.

**Review by tfrauenstein — COMMENTED** [review]
Great work -- thank you!
I proposed some editorial changes

**Comment by vadeg** [comment]
@ePaul thank you! What do you think about adding a note to the contributing guidelines about updating the changelog?

**Comment by vadeg** [comment]
👍

**Pull request merged**